src/battle/overlay_12_0224E4FC.c

```diff
@@ -3864,3 +3864,689 @@ int ov12_02253194(BattleSystem *bsys, BATTLECONTEXT *ctx) {
     
     return script;
 }
+
+int ov12_02253DA0(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+    u32 battleType;
+    int battlerIdTarget;
+    int battlerIdTargets[2];
+    int i;
+    
+    battleType = BattleSystem_GetBattleType(bsys);
+    
+    if (battleType & BATTLE_TYPE_DOUBLES) {
+        battlerIdTargets[0] = ov12_0223ABB8(bsys, battlerId, 0);
+        battlerIdTargets[1] = ov12_0223ABB8(bsys, battlerId, 2);
+        i = BattleSystem_Random(bsys) & 1;
+        battlerIdTarget = battlerIdTargets[i];
+        if (!ctx->battleMons[battlerIdTarget].hp) {
+            battlerIdTarget = battlerIdTargets[i ^ 1];
+        }
+    } else {
+        battlerIdTarget = battlerId ^ 1;
+    }
+    
+    return battlerIdTarget;
+}
+
+BOOL CheckAbilityEffectOnHit(BattleSystem *bsys, BATTLECONTEXT *ctx, int *script) {
+    BOOL ret = FALSE;
+    
+    if (ctx->battlerIdTarget == BATTLER_NONE) {
+        return ret;
+    }
+    
+    if (BattlerCheckSubstitute(ctx, ctx->battlerIdTarget) == TRUE) {
+        return ret;
+    }
+    
+    switch (GetBattlerAbility(ctx, ctx->battlerIdTarget)) {
+    case ABILITY_STATIC:
+        if (ctx->battleMons[ctx->battlerIdAttacker].hp &&
+            !ctx->battleMons[ctx->battlerIdAttacker].status &&
+            !(ctx->moveStatusFlag & MOVE_STATUS_FAIL) &&
+            !(ctx->linkStatus & 0x20) &&
```

Comment:
use shifts

Reply:
there will be shifts once I go back through and put in constants lol